_scripts/sh/create_pages.sh

```diff
@@ -25,19 +25,26 @@ _read_yaml() {
 
 read_categories() {
   local _yaml="$(_read_yaml "$1")"
-  local _categories="$(echo "$_yaml" | grep "^categories *:")"
-  local _category="$(echo "$_yaml" | grep "^category *:")"
+  local _categories="$(echo "$_yaml" | yq r - "categories.*")"
+  local _category="$(echo "$_yaml" | yq r - "category")"
 
   if [[ -n $_categories ]]; then
-    echo "$_categories" | sed "s/categories *: *//;s/\[//;s/\].*//;s/, */,/g;s/\"//g;s/'//g"
+    echo "$_categories"
   elif [[ -n $_category ]]; then
-    echo "$_category" | sed "s/category *: *//;s/\[//;s/\].*//;s/, */,/g;s/\"//g;s/'//g"
+    echo "$_category"
   fi
 }
 
 read_tags() {
   local _yaml="$(_read_yaml "$1")"
-  echo "$_yaml" | grep "^tags *:" | sed "s/tags *: *//;s/\[//;s/\].*//;s/, */,/g;s/\"//g;s/'//g"
+  local _tags="$(echo "$_yaml" | yq r - "tags.*")"
+  local _tag="$(echo "$_yaml" | yq r - "tag")"
+
+  if [[ -n $_tags ]]; then
+    echo "$_tags"
+  elif [[ -n $_tag ]]; then
+    echo "$_tag"
+  fi
```

Comment:
Sorry, there is one more thing, Jekyll only supports the `tags` variable, so this part can be reduced to:

```diff
-local _tag="$(echo "$_yaml" | yq r - "tag")"
-
-if [[ -n $_tags ]]; then
-  echo "$_tags"
-elif [[ -n $_tag ]]; then
-  echo "$_tag"
-fi
+echo "$_tags"
```

Reply:
Hmm, this [doc](https://jekyllrb.com/docs/posts/#tags) says both though.

Reply:
OK, my bad, I'm out of date :)